compiler/modules/print.llvm/llvm.bal

```diff
@@ -314,20 +323,74 @@ public class FunctionDefn {
         else {
             self.variableNames[regName] = 1;
         }
+        if name is string && !self.nameValid(regName) {
+            regName = "\"" + regName + "\"";
+        }
         string reg = "%" + regName;
         return reg;
-    } 
+    }
+
+    // Verify name is a valid LLVM name otherwise must escape
+    function nameValid(string name) returns boolean{
+        string nameLower = name.toLowerAscii();
+        // Currently check only the first character
+        string firstChar = nameLower.substring(0,1);
+        if firstChar == "_" || firstChar == "$" {
+            return true;
+        }
+        int first = firstChar.getCodePoint(0);
+        if firstChar >= "a" && firstChar <= "z" {
```

Comment:
This shouldn't compile. You can't compare an `int` and a `string`.

Reply:
Function replace with new `escapeName` function